### ISSUE TITLE
io.netty.handler.codec.CorruptedFrameException: Empty Packet!

### ISSUE THREAD

**Issue by L4BORG** [PaperMC/Paper#21]
https://github.com/SpigotMC/BungeeCord/issues/1739

is this by any chance PaperSpigot issue? Cause I'm running latest PaperSpigot and latest BungeeCord...

**Comment by Techcable** [comment]
I _highly_ doubt this is a PaperSpigot issue.
I don't see why md_5 thinks it is.

**Comment by L4BORG** [comment]
also this would occur...

```
23:14:15 [SEVERE] [/xxx.xxx.xxx.xxx:52941] <-> InitialHandler - encountered exception
io.netty.handler.codec.CorruptedFrameException: length wider than 21-bit
    at net.md_5.bungee.protocol.Varint21FrameDecoder.decode(Varint21FrameDecoder.java:67)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:360)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
    at io.netty.handler.codec.ByteToMessageDecoder.handlerRemoved(ByteToMessageDecoder.java:216)
    at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:527)
    at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved(DefaultChannelPipeline.java:521)
    at io.netty.channel.DefaultChannelPipeline.remove0(DefaultChannelPipeline.java:351)
    at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:322)
    at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:299)
    at net.md_5.bungee.protocol.LegacyDecoder.decode(LegacyDecoder.java:37)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:360)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
    at io.netty.handler.timeout.ReadTimeoutHandler.channelRead(ReadTimeoutHandler.java:152)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846)
    at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:823)
    at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:339)
    at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:255)
    at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
    at java.lang.Thread.run(Thread.java:745)
```

every now and then on BungeeCord...

**Comment by zachbr** [comment]
You are using a broken proxy.
Please try giving a less hostile proxy, like [Waterfall](https://github.com/WaterfallMC/Waterfall/), a try.

---

In all seriousness, I do not see how this is related to Paper in any way. If we get some evidence to the contrary we will of course look into it, however, I see no cause to blame Paper in this matter.

**Comment by Zbob750** [comment]
For anyone else who comes across this issue, it has been fixed where it should've been. At the proxy. Waterfall fixed this in WaterfallMC/Waterfall@a6790f3460e73e4682137fd258f172fa0b8eb753
If you have this issue, go use Waterfall. Or harass bungeecord into pulling the proper fix.
